### PR TITLE
improve commit hooks

### PR DESCRIPTION
**Pull request by smiller171**

### CODE REVIEW

**Review by sblack4 — APPROVED** [review]
:shipit:

**Review by cdaniluk — APPROVED** [review]

**Comment by smiller171** [comment]
Why are my checks failing in this repo‽

**Comment by sblack4** [comment]
```
Terraform docs...........................................................Passed
Terraform fmt............................................................Passed
terraform_tflint_deep...................................................Skipped
terraform_tflint_nocreds.................................................Failed
- hook id: terraform_tflint
- exit code: 2

Failed to initialize a runner. An error occurred:

Error: No valid credential sources found

Failed to initialize a runner. An error occurred:

Error: No valid credential sources found

Terraform validate with tfsec............................................Passed
Terraform validate.......................................................Failed
- hook id: terraform_validate
- exit code: 127

/Users/runner/.cache/pre-commit/repo89rlv4rb/terraform_validate.sh: line 20: realpath: command not found

Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Check for broken symlinks............................(no files to check)Skipped
Check Yaml...............................................................Passed
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing .pre-commit-config.yaml

Trim Trailing Whitespace.................................................Passed
Mixed line ending........................................................Passed
Don't commit to branch...................................................Passed
Pretty format JSON...................................(no files to check)Skipped
Trim Trailing Whitespace.................................................Passed
##[error]Process completed with exit code 1.
```

**Comment by smiller171** [comment]
@sblack4 Yeah I saw that much. I'm trying to figure out _why_

**Comment by sblack4** [comment]
It looks like three things;
- no AWS creds for tflint
- no `realpath`, which idk what that is
- a missing newline on the `.pre-commit-config.yaml`

IDK why it wants creds and I think I've seen the realpath error before

**Comment by sblack4** [comment]
Ahh, `realpath` is part of `coreutils` - https://github.com/antonbabenko/pre-commit-terraform#1-install-dependencies

**Comment by smiller171** [comment]
I fixed `realpath` but it's still failing, which is bullshit because it's not doing `deep`

If I totally erase my `~/.aws.credetials` tflint works in regular mode but not deep mode. So why isn't it working in Github Actions?

File: .github/workflows/pre-commit-check.yaml [hunk 1]
```diff
@@ -4,19 +4,29 @@ on:
...
-      run: |
-        brew install tfenv tflint terraform-docs pre-commit
-        pre-commit install
-        tfenv install
-    - name: pre-commit run all
-      run: |
-        pre-commit run -a
+      - uses: actions/checkout@v1
+      - name: Install prerequisites
+        run: ./bin/install-macos.sh
```
Comment (sblack4) [inline .github/workflows/pre-commit-check.yaml:16]: yes!

File: .github/workflows/pre-commit-check.yaml [hunk 2]
```diff
@@ -4,19 +4,29 @@ on:
...
+      - name: initiallize Terraform
+        run: terraform init --backend=false
+      - name: debug
+        run: tflint -v
```
Comment (sblack4) [inline .github/workflows/pre-commit-check.yaml:20]: do we want to leave debug in it? I'm fine with it, GH Actions are free for public repos

**Review by sblack4 — APPROVED** [review]
:shipit: 🔥

File: .github/workflows/pre-commit-check.yaml (hunk 2 above)
Comment (smiller171) [reply .github/workflows/pre-commit-check.yaml:20]: I think it's fine to print the version there. Can't be bothered to remove it

**Pull request merged**